common/k8s/src/middleware/metadata.rs

```diff
@@ -1,14 +1,13 @@
 use crate::errors::K8sError;
```

Comment:
Just to be clear, all the changes here are just because of the updated dependencies e.g. the behavior hasn't changed. That's what it appears to be and the logic looks sound to me when compared with the updated docs 👍

Reply:
Yup, this is purely around using the newer API version which has better types for deserializing events and a more reboust watcher mechanism